### PR TITLE
fix(build): angular now builds in non git repository

### PR DESCRIPTION
**Pull request by i3anaan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
`./build.sh`
`fatal: Not a git repository (or any of the parent directories): .git`

## What is the new behavior?
A warning is shown that says the current version cannot be determined and a placeholder (invalid) version would be used, but the build finishes.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -36,7 +36,14 @@ NODE_PACKAGES=(compiler-cli
...
 BUNDLE=true
 VERSION_PREFIX=$(node -p "require('./package.json').version")
-VERSION_SUFFIX="-$(git log --oneline -1 | awk '{print $1}')"
+
+if [ -d .git ]; then
+  VERSION_SUFFIX="-$(git log --oneline -1 | awk '{print $1}')"
+else
+  echo "WARNING: Unable to determine version from git repository status, defaulting to 0.0.0-PLACEHOLDER"
+  VERSION_SUFFIX="0.0.0-PLACEHOLDER"
+fi;
```
Comment (achembarpu) [inline build.sh]: nitpick: remove semicolon.

**Review by achembarpu — CHANGES_REQUESTED** [review]

**Review by achembarpu — APPROVED** [review]
LGTM

**Comment by i3anaan** [comment]
@Algirdyz Lacks the Linux experience to review

**Review by blaz11 — APPROVED** [review]
AMAZING

**Pull request merged**